pkg/verify/signed_entity.go

```diff
@@ -515,6 +515,18 @@ func (v *SignedEntityVerifier) Verify(entity SignedEntity, pb PolicyBuilder) (*V
 		return nil, fmt.Errorf("failed to build policy: %w", err)
 	}
 
+	// Early check: If SCTs are required, ensure the bundle is certificate-signed not public key-signed
+	if v.config.requireSCTs {
```

Comment:
Could we do this in the construction of [`SignedEntityVerifier`](https://github.com/sigstore/sigstore-go/blob/693bbb25c3404d0089ce6de3fe7bc7089a57ce14/pkg/verify/signed_entity.go#L86)?

Reply:
I agree that this implementation of binding SignedEntity and SignedEntityVerifier together isn't what we want to do. I was thinking we could implement this check by checking for a public key in the trust root material, specifically if there is a `TrustedPublicKeyMaterial` verifier. Though I recognize now this might not be straightforward - if there are multiple verifiers for both keys and certificates, it's only at verification time for the SignedEntity we will know if it is incompatible with the verifier policy. 

so tldr - let's move it to the verifier like you suggested